File: remote/src/main/java/com/kelsos/mbrc/dao/PlaylistTrack.java
package com.kelsos.mbrc.dao;

import com.kelsos.mbrc.RemoteDatabase;
import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.ForeignKey;
import com.raizlabs.android.dbflow.annotation.ForeignKeyReference;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.structure.BaseModel;

@Table(databaseName = RemoteDatabase.NAME) public class PlaylistTrack extends BaseModel {
  @Column @PrimaryKey(autoincrement = true) long id;
  @Column private int position;
  @Column private String path;
  @Column private String artist;
  @Column private String title;
  @Column @ForeignKey(
      references = {
          @ForeignKeyReference(columnName = "playlist_id",
              columnType = Long.class,
              foreignColumnName = "id")
      }, saveForeignKeyModel = false) private PlaylistDao playlist;

  public PlaylistDao getPlaylist() {
    return playlist;
  }

  public PlaylistTrack setPlaylist(PlaylistDao playlist) {
    this.playlist = playlist;
    return this;
  }

  public String getTitle() {
    return title;
  }

  public PlaylistTrack setTitle(String title) {
    this.title = title;
    return this;
  }

  public String getArtist() {
    return artist;
  }

  public PlaylistTrack setArtist(String artist) {
    this.artist = artist;
    return this;
  }

  public String getPath() {
    return path;
  }

  public PlaylistTrack setPath(String path) {
    this.path = path;
    return this;
  }

  public int getPosition() {
    return position;
  }

  public PlaylistTrack setPosition(int position) {
    this.position = position;
    return this;
  }

  public long getId() {
    return id;
  }

  public PlaylistTrack setId(long id) {
    this.id = id;
    return this;
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/data/dao/PlaylistTrack.kt
package com.kelsos.mbrc.data.dao

import com.kelsos.mbrc.data.db.RemoteDatabase
import com.raizlabs.android.dbflow.annotation.*
import com.raizlabs.android.dbflow.kotlinextensions.modelAdapter
import com.raizlabs.android.dbflow.structure.Model

@Table(database = RemoteDatabase::class, name = "playlist_track")
data class PlaylistTrack(
    @Column var position: Int = 0,
    @Column
    @ForeignKey(references = arrayOf(ForeignKeyReference(columnName = "playlist_id",
        foreignKeyColumnName = "id")), saveForeignKeyModel = false)
    var playlist: Playlist? = null,
    @Column
    @ForeignKey(references = arrayOf(ForeignKeyReference(columnName = "track_info_id",
        foreignKeyColumnName = "id")), saveForeignKeyModel = false)
    var trackInfo: PlaylistTrackInfo? = null,
    @Column(name = "date_added") var dateAdded: Long = 0,
    @Column(name = "date_updated") var dateUpdated: Long = 0,
    @Column(name = "date_deleted") var dateDeleted: Long = 0,
    @Column @PrimaryKey(autoincrement = true) var id: Long = 0
) : Model {

  override fun load() = modelAdapter<PlaylistTrack>().load(this)

  override fun insert(): Long = modelAdapter<PlaylistTrack>().insert(this)

  override fun save() = modelAdapter<PlaylistTrack>().save(this)

  override fun update() = modelAdapter<PlaylistTrack>().update(this)

  override fun exists(): Boolean = modelAdapter<PlaylistTrack>().exists(this)

  override fun delete() = modelAdapter<PlaylistTrack>().delete(this)
}
